parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -925,24 +884,7 @@ def __parse_dependents(self):
                     for module in self.__existing_modules_dict.values():
                         if module['name'] == search['name']:
                             rev = search.get('revision')
-                            if rev is not None:
-                                if rev == module['revision']:
-                                    new = {'name': name,
-                                           'revision': revision,
-                                           'schema': mod['schema']}
-                                    if module.get('dependents') is None:
-                                        module['dependents'] = []
-                                    if new not in module['dependents']:
-                                        module['dependents'].append(new)
-                                        name_revision = '{}@{}'.format(module['name'], module['revision'])
-                                        if self.__new_modules.get(name_revision) is None:
-                                            self.__new_modules[name_revision] = module
-                                        else:
-                                            if self.__new_modules[name_revision].get('dependents') is None:
-                                                self.__new_modules[name_revision]['dependents'] = module['dependents']
-                                            else:
-                                                self.__new_modules[name_revision]['dependents'].append(new)
-                            else:
+                            if rev is None or rev == module['revision']:
```

Comment:
**NOTE**: This condition was re-evaluated so the duplicate code could be merged together